vault-admin-client/src/main/java/com/scality/osis/vaultadmin/impl/VaultAdminImpl.java

```diff
@@ -6,50 +6,63 @@
 package com.scality.osis.vaultadmin.impl;
 
 import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest;
 import com.scality.osis.vaultadmin.impl.cache.Cache;
 import com.scality.osis.vaultadmin.impl.cache.CacheConstants;
 import com.scality.osis.vaultadmin.impl.cache.CacheFactory;
+import com.scality.vaultclient.dto.AssumeRoleResult;
 import com.scality.vaultclient.dto.CreateAccountRequestDTO;
 import com.scality.vaultclient.dto.CreateAccountResponseDTO;
 import com.scality.vaultclient.dto.ListAccountsRequestDTO;
 import com.scality.vaultclient.dto.ListAccountsResponseDTO;
 import com.scality.vaultclient.services.AccountServicesClient;
-import okhttp3.*;
+import com.scality.vaultclient.services.SecurityTokenServicesClient;
 import com.scality.osis.vaultadmin.VaultAdmin;
+import okhttp3.HttpUrl;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import com.amazonaws.services.securitytoken.model.Credentials;
 
 /**
  * Vault administrator implementation
  *
  * <p>Created by saitharun on 12/16/20.
  */
-@SuppressWarnings("deprecation")
+@SuppressWarnings({"deprecation", "unchecked"})
 public class VaultAdminImpl implements VaultAdmin{
   private static final Logger logger = LoggerFactory.getLogger(VaultAdminImpl.class);
 
   public static final String CD_TENANT_ID_PREFIX = "cd_tenant_id";
 
   private final AccountServicesClient vaultAccountClient;
 
+  private final SecurityTokenServicesClient vaultSTSClient;
+
   @Autowired
   private CacheFactory cacheFactory;
 
   private Cache<Integer, String> listAccountsMarkerCache;
 
+  private Cache<String, Credentials> assumeRoleCache;
+
   /**
    * Create a Vault administrator implementation
    *  @param accessKey Access key of the admin who have proper administrative capabilities.
    * @param secretKey Secret key of the admin who have proper administrative capabilities.
    * @param endpoint Vault admin API endpoint, e.g., http://127.0.0.1:80/admin
    */
-  public VaultAdminImpl(String accessKey, String secretKey, String endpoint) {
+  public VaultAdminImpl(String accessKey, String secretKey, String endpoint, String s3Endpoint) {
     validEndpoint(endpoint);
     this.vaultAccountClient = new AccountServicesClient(
             new BasicAWSCredentials(accessKey, secretKey));
     vaultAccountClient.setEndpoint(endpoint);
 
+
+    this.vaultSTSClient = new SecurityTokenServicesClient(
+            new BasicAWSCredentials(accessKey, secretKey));
+    vaultSTSClient.setEndpoint(s3Endpoint);
```

Comment:
Shouldn't it be namedd STS Endpoint.
`<Vault-IP>:8800`

Reply:
No `AssumeRoleBackbeat` is called with s3 port 8500 so s3 URl

Reply:
Please let me know if we can invoke route `AssumeRoleBackbeat` with port 8800

Reply:
Are you making a call to S3 ie. clousderver and then asking s3 to make a call to vault on port 8500?
If yes, then good. If OSIS is making a call directly on port 8500, then its vault.

Yes, you are right, its port 8500.

Reply:
fixed it in 916eec1d0143504a70d18803cb68a1a3643cabaa